text/src/swam/witx/unresolved/Types.scala

```diff
@@ -0,0 +1,36 @@
+package swam
+package witx
+package unresolved
+
+/**
+@author Javier Cabrera-Arteaga on 2020-03-18
+  */
+sealed trait BaseWitxType
+
+case class BasicType(name: String) extends BaseWitxType
+
+case class AliasType(name: String) extends BaseWitxType
+
+case class EnumType(`type`: BaseWitxType, names: Seq[String]) extends BaseWitxType
+
+case class FlagsType(`type`: BaseWitxType, names: Seq[String]) extends BaseWitxType
+
+case class ArrayType(`type`: BaseWitxType) extends BaseWitxType
+
+case class Pointer(`type`: BaseWitxType) extends BaseWitxType
+
+case class StructType(fields: Seq[Field]) extends BaseWitxType
```

Comment:
Can you avoid using `Seq` and use the concrete `List` type instead?

Reply:
Gotcha